tensorboard/plugins/image/summary.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Image summaries and TensorFlow operations to create them.
+
+An image summary stores the width, height, and PNG-encoded data for zero
+or more images in a rank-1 string array: `[w, h, png0, png1, ...]`.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow as tf
+
+from tensorboard.plugins.image import metadata
+
+
+def op(name,
+       images,
+       max_outputs=3,
+       display_name=None,
+       description=None,
+       collections=None):
+  """Create an image summary op for use in a TensorFlow graph.
+
+  Arguments:
+    name: A unique name for the generated summary node.
+    images: A `Tensor` representing pixel data with shape `[k, w, h, c]`,
+      where `k` is the number of images, `w` and `h` are the width and
+      height of the images, and `c` is the number of channels, which
+      should be 1, 3, or 4. Any of the dimensions may be statically
+      unknown (i.e., `None`).
+    max_outputs: Optional `int` or rank-0 integer `Tensor`. At most this
+      many images will be emitted at each step. When more than
+      `max_outputs` many images are provided, the first `max_outputs` many
+      images will be used and the rest silently discarded.
+    display_name: Optional name for this summary in TensorBoard, as a
+      constant `str`. Defaults to `name`.
+    description: Optional long-form description for this summary, as a
+      constant `str`. Markdown is supported. Defaults to empty.
+    collections: Optional list of graph collections keys. The new
+      summary op is added to these collections. Defaults to
+      `[Graph Keys.SUMMARIES]`.
+
+  Returns:
+    A TensorFlow summary op.
+  """
+  if display_name is None:
+    display_name = name
+  summary_metadata = metadata.create_summary_metadata(
+      display_name=display_name, description=description)
+  with tf.name_scope(name), \
+       tf.control_dependencies([tf.assert_rank(images, 4),
+                                tf.assert_type(images, tf.uint8),
```

Comment:
What happened to that gist you made before, where you recreated the C++ image summary code as a TF graph? We probably shouldn't have any regressions in terms of behavior, if it can be avoided.